thundra/context/tracing_execution_context_provider.py

```diff
@@ -0,0 +1,24 @@
+from thundra.context.execution_context import ExecutionContext
+from thundra.opentracing.tracer import ThundraTracer
+
+
+class TracingExecutionContextProvider:
+    def __init__(self):
+        self.tracer = ThundraTracer.get_instance()
+        self.execution_context = None
+
+    def get(self):
+        execution_context = self.execution_context
+        active_span = self.tracer.get_active_span()
+        if active_span and hasattr(active_span, 'execution_context'):
+            execution_context = active_span.execution_context
+        if not execution_context:
+            return ExecutionContext()
+        return execution_context
+    
+    def set(self, execution_context):
+        self.execution_context = execution_context
```

Comment:
what is the use case for setting exec context of `TracingExecutionContextProvider`? During tests?

Reply:
It is not actually used in anywhere (tests use globalExecutionExecutionContextProvider but we might add tests for TracingExecutionContextProvider too)

Reply:
I think, it should set/clear exec context in active span and should not have a global exec context prop